crates/runtime/src/lib.rs

```diff
@@ -610,15 +658,72 @@ impl Runtime {
     }
 
     // Closes and deallocates all resources (including the static registries)
-    pub async fn shutdown(self) {
+    pub async fn shutdown(&self) {
+        if self.status.is_shutdown() {
+            return;
+        }
+
+        tracing::info!("Shutting down runtime...");
         self.status.mark_shutdown();
+
+        // shutdown all running components except the HTTP and Metrics servers
+        let mut running_components = self.server_components.write().await;
+
+        // HTTP and METRICS servers must be shutdown last
+        let (to_shutdown, to_shutdown_last) = running_components
+            .drain()
+            .partition::<Vec<_>, _>(|(name, _)| *name != HTTP_SERVER && *name != METRICS_SERVER);
```

Comment:
```suggestion
        // HTTP and METRICS servers must be shutdown last
        let mut first_shutdown_group = Vec::new();
        let mut last_shutdown_group = Vec::new();
        
        for (name, handle) in running_components.drain() {
            match name.as_str() {
                HTTP_SERVER | METRICS_SERVER => last_shutdown_group.push((name, handle)),
                _ => first_shutdown_group.push((name, handle)),
            }
        }
```

I think this is slightly cleaner/easier to understand and allows us to create more groups in the future if we need it.

Reply:
Thank you, updated